CRM/Pcpteams/Form/TeamJoin.php

```diff
@@ -11,6 +11,10 @@ static function preProcess(&$form) {
     if (!$form->get('page_id')) {
       CRM_Core_Error::fatal(ts("Can't determine pcp id."));
     }
+    if(!$form->get('component_page_id')){
+     $componentPageId = CRM_Utils_Request::retrieve('pageId', 'Positive', CRM_Core_DAO::$_nullArray, TRUE);
+     $form->set('component_page_id', $componentPageId);
+    }
```

Comment:
This is already taken care in workflow.php - preProcess(). Sth else wrong with url / workflow.